File: passiveScalarTransport2D/corrMom2D.jl
function corrMom2D!(rhou, rhouP, p, dt, delta)
	# corrects momentum prediction field based on given pressure field
	# use CDS for pressure gradient

	nG = 1
	ImaAll, JmaAll = size(rhou)
	Ifi = nG +1; Ifim = Ifi-1; Ifip = Ifi +1
	Ila = ImaAll - nG; Ilap = Ila+1; Ilam = Ila-1
	Jfi = nG +1; Jfim = Jfi-1; Jfip = Jfi +1
	Jla = JmaAll - nG; Jlap = Jla+1; Jlam = Jla-1

	rhou[Ifi:Ila, Jfi:Jla, 1] = rhouP[Ifi:Ila, Jfi:Jla, 1] .- dt * 0.5 /delta *(p[Ifip:Ilap, Jfi:Jla] .-p[Ifim:Ilam, Jfi:Jla]) 

	rhou[Ifi:Ila, Jfi:Jla, 2] = rhouP[Ifi:Ila, Jfi:Jla, 2] .- dt * 0.5 /delta *(p[Ifi:Ila, Jfip:Jlap] .-p[Ifi:Ila, Jfim:Jlam]) 
	
	return nothing
end

File: nfs2024_matlab/simpleFlowSolver2D/corrMom2D.m
% # net.m ###############################################
% Programm: 		    	simpleFlowSolver2D
% Content:					apply p corr to Mom field
% Author:                   S.H. Tey, June 2024
% Email:                    sheu.tey@stud.uni-due.de
% Matriculation Number:     3117568
% -------------------------------------------------------
% function to apply momentum field correction with 
% pressure gradient
% =======================================================

function rhou = corrMom2D(rhou, rhouP, p)% ############
% ------- parameters ------------------------------------
global Ifi Ifim Ifip Ila Ilam Ilap 
global Jfi Jfim Jfip Jla Jlam Jlap
global Delta DeltaT
% ------- preallocating and preindexing -----------------
pE = p(Ifip:Ilap, Jfi:Jla);
pW = p(Ifim:Ilam, Jfi:Jla);
pN = p(Ifi:Ila, Jfip:Jlap);
pS = p(Ifi:Ila, Jfim:Jlam);
rhouPX = rhouP(Ifi:Ila, Jfi:Jla, 1);
rhouPY = rhouP(Ifi:Ila, Jfi:Jla, 2);
% ------- operation -------------------------------------
rhou(Ifi:Ila, Jfi:Jla, 1) = rhouPX - DeltaT *0.5 ...
    /Delta * ( pE - pW );
rhou(Ifi:Ila, Jfi:Jla, 2) = rhouPY - DeltaT *0.5 ...
    /Delta * ( pN - pS );
end % ###################################################
